xroad-catalog-collector/src/main/java/fi/vrk/xroad/catalog/collector/actors/FetchOrganizationsActor.java

```diff
@@ -0,0 +1,282 @@
+/**
+ * The MIT License
+ * Copyright (c) 2016, Population Register Centre (VRK)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package fi.vrk.xroad.catalog.collector.actors;
+
+import fi.vrk.xroad.catalog.collector.util.OrganizationUtil;
+import fi.vrk.xroad.catalog.collector.wsimport.ClientList;
+import fi.vrk.xroad.catalog.persistence.entity.*;
+import fi.vrk.xroad.catalog.persistence.CatalogService;
+import lombok.extern.slf4j.Slf4j;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+
+/**
+ * Actor which fetches organizations with respective data
+ */
+@Component
+@Scope("prototype")
+@Slf4j
+public class FetchOrganizationsActor extends XRoadCatalogActor {
+
+    @Autowired
+    protected CatalogService catalogService;
+
+    @Override
+    public void preStart() throws Exception {
+
+    }
+
+    @Override
+    protected boolean handleMessage(Object message) throws NoSuchAlgorithmException, KeyStoreException, KeyManagementException {
+        if (message instanceof ClientList) {
+            String organizationListUrl = "https://api.palvelutietovaranto.suomi.fi";
```

Comment:
Hard coded URL.

Reply:
Fixed

Reply:
Actually, one thing came to my mind: Does the property fetchOrganizationsHost have the same content as the previous organizationListUrl? If yes, I would prefer property ending Url since it is not only host name but the whole url.

Reply:
Fixed